Added support to cache the value of the id32bit test, to allow cross compilation.

	--xtraeme

--- configure.in.orig	2014-05-09 13:53:10.000000000 +0200
+++ configure.in	2014-08-25 10:41:31.198567162 +0200
@@ -335,6 +335,9 @@ if test "$enable_subids" != "no"; then
 	dnl
 	dnl FIXME: check if 32 bit UIDs/GIDs are supported by libc
 	dnl
+	AC_CACHE_CHECK(
+		[wheter subordinate IDs are supported],
+		[ac_cv_id32bit], [
 	AC_RUN_IFELSE([AC_LANG_SOURCE([
 #include <sys/types.h>
 int main(void) {
@@ -342,9 +345,9 @@ int main(void) {
 	gid_t g;
 	return (sizeof u < 4) || (sizeof g < 4);
 }
-	])], [id32bit="yes"], [id32bit="no"])
+	])])], [ac_cv_id32bit="yes"], [ac_cv_id32bit="no"])
 
-	if test "x$id32bit" = "xyes"; then
+	if test "x$ac_cv_id32bit" = "xyes"; then
 		AC_DEFINE(ENABLE_SUBIDS, 1, [Define to support the subordinate IDs.])
 		enable_subids="yes"
 	else
